name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -1246,59 +1255,50 @@ private void addDividendMarkerLines(ChartInterval chartInterval)
             if (chartConfig.contains(ChartDetails.SHOW_DATA_LABELS))
             {
                 customPaintListeners.add(event -> {
+                    FontMetrics fontMetrics = event.gc.getFontMetrics();
+
+                    StringBuilder labelBuilder = new StringBuilder();
+
+                    // Three levels of the label
+                    int[] yPosLabels = new int[3];
```

Comment:
You do not need to store the Y position - it is fixed. Anyway, you only write into the array just before reading out of the array. Therefore it can be a local variable.